python/eos/analysis_file_description.py

```diff
@@ -0,0 +1,309 @@
+            return Deserializable.make(ConstraintPriorDescription, **kwargs)
+        elif 'parameter' in kwargs and 'type' in kwargs:
+            _kwargs = _copy.deepcopy(kwargs)
+            _kwargs.pop('type')
+            if kwargs['type'] in ("uniform", "flat"):
+                return Deserializable.make(UniformPriorDescription, **_kwargs)
+            elif kwargs['type'] in ("scale",):
+                return Deserializable.make(ScalePriorDescription, **_kwargs)
+            elif kwargs['type'] in ("gauss", "gaussian"):
+                if "min" in kwargs:
+                    return Deserializable.make(CurtailedGaussianDescription, **_kwargs)
+                return Deserializable.make(GaussianPriorDescription, **_kwargs)
+            elif kwargs['type'] in ("poisson",):
+                return Deserializable.make(PoissonPriorDescription, **_kwargs)
+
+
+        raise ValueError('Unknown type of prior description')
+
+@dataclass
+class PoissonPriorDescription(Deserializable):
+    parameter:str
+    k:float
+    type: str = field(repr=False, init=False, default="poisson")
+
+@dataclass
+class CurtailedGaussianDescription(Deserializable):
+    parameter:str
+    central:float
+    sigma:float
+    min:float
+    max:float
+    type: str = field(repr=False, init=False, default="curtailed gaussian")
+
+@dataclass
+class GaussianPriorDescription(Deserializable):
+    parameter: str
+    central: float
+    sigma: float
+    type: str = field(repr=False, init=False, default="gaussian")
+
+
+@dataclass
+class ScalePriorDescription(Deserializable):
+    parameter: str
+    min: float
+    max: float
+    mu_0: float
+    lambda_scale: float
+    type: str = field(repr=False, init=False, default="scale")
+
+@dataclass
+class UniformPriorDescription(Deserializable):
+    parameter: str
+    min: float
+    max: float
+    type: str = field(repr=False, init=False, default="uniform")
+
+@dataclass
+class ConstraintPriorDescription(Deserializable):
+     constraint: str
+
+@dataclass
+class PriorComponent(Deserializable):
+    name: str
+    descriptions: list
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        _kwargs = _copy.deepcopy(kwargs)
+        if "descriptions" in kwargs:
+            if "parameters" in kwargs:
+                eos.error(f'Both \'descriptions\' and \'parameters\' are provided for prior component \'{kwargs["name"]}\', ignoring legacy support for \'parameters\'')
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['descriptions']]
+        if "parameters" in kwargs:
+            eos.warn(f'\'parameters\' is in the description of prior component \'{kwargs["name"]}\', use \'descriptions\' instead')
+            _kwargs.pop("parameters")
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['parameters']]
+        return Deserializable.make(cls, **_kwargs)
+
+
+
+@dataclass
+class EOSConstraintDescription(Deserializable):
```

Comment:
Hm. Do not like the ``EOS`` prefix here. Why not ``ConstraintLikelihoodDescription``?

Reply:
I chose this as these are pre-existing EOS constraints, so it is literally a description of an EOS constraint. But I didn't do that for the EOS priors so :shrug:. It would probably make sense for `EOSConstraintDescription` and `ConstraintPriorDescription` to be named consistently though. I would go for `EOSConstraintDescription` and `EOSConstraintPriorDescription`, but `ConstraintLikelihoodDescription` and `ConstraintPriorDescription` work fine as well.

Reply:
I would remove ``EOS`` from the name altogether and prefix with ``Manual`` in cases where we add information not contained in the constraints database / parameter database.